apps/cowswap-frontend/src/modules/bridge/containers/BridgeRouteBreakdown/index.tsx

```diff
@@ -98,63 +101,127 @@ export function BridgeRouteBreakdown({
   isBridgeSectionCollapsible = false,
   isBridgeSectionExpanded = true,
   onBridgeSectionToggle = () => {},
+  winningSolverId,
+  receivedAmount = null,
+  surplusAmount = null,
+  swapExplorerUrl,
+  bridgeExplorerUrl,
 }: BridgeRouteBreakdownProps) {
   const sellToken = sellCurrencyAmount.currency
   const buyToken = buyCurrencyAmount.currency
 
-  const derivedSourceChainId = sellToken.chainId as SupportedChainId
+  // Derive chain information
+  // This only needs to be recomputed when the chain ID changes
+  const sourceChainData = useMemo(() => {
+    const chainId = sellToken.chainId as SupportedChainId
+    const info = getChainInfo(chainId)
+    return {
+      chainId,
+      chainInfo: info,
+      chainName: info.label,
+    }
+  }, [sellToken.chainId])
+
+  const destChainData = useMemo(() => {
+    const chainId = bridgeReceiveCurrencyAmount.currency.chainId as SupportedChainId
+    const info = getChainInfo(chainId)
```

Comment:
It won't work. `chainId` might not be `SupportedChainId` and an error will be thrown here.